server/backend/ios_notifications/ios_apns/ios_apns_jwt/iosAPNsToken.go

```diff
@@ -0,0 +1,148 @@
+// Package ios_apns_jwt handles the generation and validation of the JWT token for the APNs service
+package ios_apns_jwt
+
+import (
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"github.com/golang-jwt/jwt"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+const (
+	// TokenTimeout for the token in seconds
+	TokenTimeout = 3000
+)
+
+var (
+	ErrorAuthKeyNotPem   = errors.New("failed to parse token: AuthKey must be a valid .p8 PEM file")
+	ErrorAuthKeyNotECDSA = errors.New("failed to parse token: AuthKey must be of type ecdsa.PrivateKey")
+	ErrorAuthKeyNil      = errors.New("failed to parse token: AuthKey was nil")
+	APNsKeyId            = os.Getenv("APNS_KEY_ID")
+	APNsTeamId           = os.Getenv("APNS_TEAM_ID")
+	APNsP8FilePath       = os.Getenv("APNS_P8_FILE_PATH")
+)
+
+type Token struct {
+	sync.Mutex
+	EncryptionKey *ecdsa.PrivateKey
+	KeyId         string
+	TeamId        string
+	IssuedAt      int64
+	Bearer        string
+}
+
+func NewToken() (*Token, error) {
+	encryptionKey, err := APNsEncryptionKeyFromFile()
+
+	if err != nil {
+		return nil, err
+	}
+
+	token := Token{
+		EncryptionKey: encryptionKey,
+		KeyId:         APNsKeyId,
+		TeamId:        APNsTeamId,
+	}
+
+	_, err = token.Generate()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &token, nil
+}
+
+// APNsEncryptionKeyFromFile reads the APNs encryption key from the file system
```

Comment:
```suggestion
// Reads the APNs encryption key from the file system
```

Don't start of the explanation with a function name, since it is known anyway.